File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
class Solution {
public:
    int trap(vector<int>& ht) {
        vector<int> lmax;
        vector<int> rmax;
        int l=ht[0];
        int r=ht[ht.size()-1];

        for(int i=0; i<ht.size(); i++){
            l=max(l, ht[i]);
            lmax.push_back(l);
            r=max(r, ht[ht.size()-1-i]);
            rmax.push_back(r);
        }
        // for(int i: lmax){cout<<i;}
        // cout<<endl;
        reverse(rmax.begin(), rmax.end());
        // for(int i: rmax){cout<<i;}111
        int res=0;
        for(int i=0; i<ht.size(); i++){
            res+=min(lmax[i], rmax[i])-ht[i];
        }

        return res;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:
        res=[0]
        def back(c):
            if(c==n):
                res[0]+=1
                return
            if(c>n):
                return
            back(c+1)
            back(c+2)

        back(0)
        return res[0]